appinventor/blocklyeditor/src/blocks/dictionaries.js

```diff
@@ -0,0 +1,483 @@
+  init: function() {
+    this.setColour(Blockly.DICTIONARY_CATEGORY_HUE);
+    this.setOutput(true, Blockly.Blocks.Utilities.YailTypeToBlocklyType("any",Blockly.Blocks.Utilities.OUTPUT));
+    var checkTypeDict = Blockly.Blocks.Utilities.YailTypeToBlocklyType("dictionary",Blockly.Blocks.Utilities.INPUT);
+    var checkTypeAny = Blockly.Blocks.Utilities.YailTypeToBlocklyType("any",Blockly.Blocks.Utilities.INPUT);
+    var checkTypeKey = Blockly.Blocks.Utilities.YailTypeToBlocklyType("key",Blockly.Blocks.Utilities.INPUT);
+    this.interpolateMsg(Blockly.Msg.LANG_DICTIONARIES_DICTIONARY_LOOKUP_INPUT,
+      ['KEY', checkTypeKey, Blockly.ALIGN_RIGHT],
+      ['DICT', checkTypeDict, Blockly.ALIGN_RIGHT],
+      ['NOTFOUND', checkTypeAny, Blockly.ALIGN_RIGHT],
+      Blockly.ALIGN_RIGHT);
+    this.setTooltip(Blockly.Msg.LANG_DICTIONARIES_DICTIONARY_LOOKUP_TOOLTIP);
+    this.setInputsInline(false);
+  },
+  typeblock: [{ translatedName: Blockly.Msg.LANG_DICTIONARIES_DICTIONARY_LOOKUP_TITLE }]
+};
+
+Blockly.Blocks['dictionaries_set_pair'] = {
+  category: 'Dictionaries',
+  helpUrl: Blockly.Msg.LANG_DICTIONARIES_SET_PAIR_HELPURL,
+  init: function() {
+    this.setColour(Blockly.DICTIONARY_CATEGORY_HUE);
+    var checkTypeDict = Blockly.Blocks.Utilities.YailTypeToBlocklyType("dictionary",Blockly.Blocks.Utilities.INPUT);
+    var checkTypeKey = Blockly.Blocks.Utilities.YailTypeToBlocklyType("key",Blockly.Blocks.Utilities.INPUT);
+    this.interpolateMsg(Blockly.Msg.LANG_DICTIONARIES_SET_PAIR_INPUT,
+            ['KEY', checkTypeKey, Blockly.ALIGN_RIGHT],
+            ['DICT', checkTypeDict, Blockly.ALIGN_RIGHT],
+            ['VALUE', null, Blockly.ALIGN_RIGHT],
+            Blockly.ALIGN_RIGHT);
+    this.setPreviousStatement(true);
+    this.setNextStatement(true);
+    this.setTooltip(Blockly.Msg.LANG_DICTIONARIES_SET_PAIR_TOOLTIP);
+    this.setInputsInline(false);
+  },
+  typeblock: [{ translatedName: Blockly.Msg.LANG_DICTIONARIES_SET_PAIR_TITLE }]
+};
+
+Blockly.Blocks['dictionaries_delete_pair'] = {
+  category: 'Dictionaries',
+  helpUrl: Blockly.Msg.LANG_DICTIONARIES_DELETE_PAIR_HELPURL,
+  init: function() {
+    this.setColour(Blockly.DICTIONARY_CATEGORY_HUE);
+    var checkTypeDict = Blockly.Blocks.Utilities.YailTypeToBlocklyType("dictionary",Blockly.Blocks.Utilities.INPUT);
+    var checkTypeKey = Blockly.Blocks.Utilities.YailTypeToBlocklyType("key",Blockly.Blocks.Utilities.INPUT);
+    this.interpolateMsg(Blockly.Msg.LANG_DICTIONARIES_DELETE_PAIR_INPUT,
+            ['DICT', checkTypeDict, Blockly.ALIGN_RIGHT],
+            ['KEY', checkTypeKey, Blockly.ALIGN_RIGHT],
+            Blockly.ALIGN_RIGHT);
+    this.setPreviousStatement(true);
+    this.setNextStatement(true);
+    this.setTooltip(Blockly.Msg.LANG_DICTIONARIES_DELETE_PAIR_TOOLTIP);
+    this.setInputsInline(false);
+  },
+  typeblock: [{ translatedName: Blockly.Msg.LANG_DICTIONARIES_DELETE_PAIR_TITLE }]
+};
+
+Blockly.Blocks['dictionary_recursive_lookup'] = {
```

Comment:
Some names start with 'dictionary' while others start with 'dictionaries'. Can you justify their usage? If it's not because of technical reasons, we should name them uniformly.

Reply:
@pavi2410 Good catch. I'll make them all plural to match the fact that we use `lists_` as a prefix for list operations.